pkgs/games/wesnoth/default.nix

```diff
@@ -1,34 +1,118 @@
-{ lib, stdenv, fetchFromGitHub
-, cmake, pkg-config, SDL2, SDL2_image, SDL2_mixer, SDL2_net, SDL2_ttf
-, pango, gettext, boost, libvorbis, fribidi, dbus, libpng, pcre, openssl, icu
-, lua, curl
-, Cocoa, Foundation
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  SDL2,
+  SDL2_image,
+  SDL2_mixer,
+  SDL2_net,
+  SDL2_ttf,
+  pango,
+  gettext,
+  boost,
+  libvorbis,
+  fribidi,
+  dbus,
+  libpng,
+  pcre,
+  openssl,
+  icu,
+  lua,
+  curl,
+  gsl-lite,
+  nix-update-script,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "wesnoth";
   version = "1.18.3";
 
   src = fetchFromGitHub {
-    rev = version;
     owner = "wesnoth";
     repo = "wesnoth";
+    tag = finalAttrs.version;
     hash = "sha256-Uk8omtXYZaneyBr4TASRtIKEyJLGwfKWu9vRQNVpdVA=";
   };
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  patches = lib.optionals stdenv.cc.isClang [
```

Comment:
Since this is llvm-19 specific, can we do a more precise version check here? Since nixpkgs allows using both [earlier and later versions](https://github.com/NixOS/nixpkgs/blob/529f6e094fa1ca92ae250f010a65f846474a8c3c/pkgs/development/compilers/llvm/default.nix#L32), it may be nicer to only apply this for llvm19+

Reply:
Fair enough, will do!